src/robotFactory.js

```diff
@@ -1,10 +1,90 @@
 'use strict';
 
-class BaseRobot {}
+const BASIC_COORDS_X_Y = {
+  x: 0,
+  y: 0,
+};
+
+const BASIC_COORDS_X_Y_Z = {
+  x: 0,
+  y: 0,
+  z: 0,
+};
+
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+
+    this.coords = {
+      ...BASIC_COORDS_X_Y,
+      ...coords,
+    };
```

Comment:
Interesting solution!
You can also do it like this:
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```